sources/api/apiserver/src/server/mod.rs

```diff
@@ -543,6 +545,49 @@ async fn reboot() -> Result<HttpResponse> {
     Ok(HttpResponse::NoContent().finish())
 }
 
+/// Triggers report generation for the requested report type and returns the results.
+async fn get_report(query: web::Query<HashMap<String, String>>) -> Result<HttpResponse> {
+    trace!("Generating report");
+
+    match query.get("type") {
+        Some(report_type) => match report_type.as_str() {
+            "cis" => get_cis_report(&query).await,
```

Comment:
How long does `bloodhound` take to generate the CIS report? We want to minimize the time `apiserver` blocks on any API request.

Reply:
For the most part, the report generation is just checking the content of a handful of files. So the CIS report is actually pretty quick:

```
real    0m0.079s
user    0m0.043s
sys     0m0.008s
```

We may eventually want to add other types of reports. These will all be exposed through `apiclient` under a common `report` subcommand. If/when we add a type of report that would take more time, the plan is `apiclient` logic will abstract away the details and call an async endpoint under the hood.

So with the current command, `apiclient report cis` would end up calling `GET /report&type=cis`.

If we add something more resource intesive, we would have something like `apiclient report io-sampling` that would end up routing things through something like `POST /actions/report body: {"type": "io-sample"}`, then poll for the results from something like `GET /actions/report-result body {"id": $async_id}`.

Specific details to be figured out if/when we go there, of course. But for now, this is so quick that I figured adding async request/result handling would just add extra overhead on both the client and server ends.